### PR TITLE
Automated tests for Feature Index Pages

### LINKED ISSUE #7824: Create FeatureIndexPage from IdxPage
**Is your feature request related to a problem? Please describe.**
Feature Index pages are currently using the FrontPage container, but should have its own page type / container

**Describe the solution you'd like**
- [ ] Copy Feature Index Page functionality from `IdxPage` container into top level `FeatureIndexPage` container
- [ ] Adjust routing here https://github.com/bbc/simorgh/blob/latest/src/app/routes/cpsAsset/index.jsx#L24-L29
- [ ] Refactor / reuse logic / components within both `FrontPage` / `IdxPage` containers where possible / appropriate

**Describe alternatives you've considered**

**Testing notes**
[Tester to complete]

Dev insight: Will Cypress tests be required or are unit tests sufficient? Will there be any potential regression? etc

- [ ] This feature is expected to need manual testing.

**Additional context**

**Questions**
- [ ] Currently, if there is no strapline, the IndexPageSection will not render https://github.com/bbc/simorgh/blob/latest/src/app/containers/IndexPageSection/index.jsx#L187-L193. We have removed this check, but need to confirm a11y / product / editorial that this is OK? This will have an effect on FrontPages & IdxPages.
  - Additional padding on the index divider if no strapline
- [ ] On PAL, the number of promos to display within a group is limited - should this also be the case for Simorgh? 
- [ ] Most Read was out of scope, but we have included it on the FIX page - confirm that this is OK
- [ ] Should FIX Pages have a secondary column?
- [ ] There was a requirement to "then flatten this section and add to end of the previous section" - is this still the case? If we did this, all promos would be within Top Stories...

### PR DESCRIPTION
**Pull request by karinathomasbbc**
Resolves #7824

**Overall change:**
We need to add automated tests for Feature Index Pages e.g. a11y & AMP validation

**Code changes:**
- Enable a11y tests for feature index pages
- Enable ampValidation for feature index pages

---

- [ ] I have assigned myself to this PR and the corresponding issues
- [ ] I have added the `cross-team` label to this PR if it requires visibility across World Service teams
- [ ] I have assigned this PR to the Simorgh project

**Testing:**

- [ ] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false npm run test:e2e:interactive`)
- [ ] This PR requires manual testing

### CODE REVIEW

File: a11y.js [hunk 1]
```diff
@@ -67,6 +67,7 @@ const pageTypes = {
     '/iframe', // same issues as in mediaEmbedErrorsToSuppress but the DOM path is different
     "//div[@id='root']/div/div[1]/main/div[37]/div/div/div", // issue with IDT2 includes
     "//div[@id='ns_datapic_royal-engagements']",
+    "//div[@id='ns_datapic_env-test-2']",
```
Comment (simonsinclair) [inline a11y.js:77]: What does this change do?

File: a11y.js [hunk 2]
```diff
@@ -77,13 +78,19 @@ const pageTypes = {
   idxPage: ["//div[@id='root']/header/nav/div/div[1]/div/ul"],
   onDemandTV: ['/iframe'],
   mediaAssetPage: [...mediaEmbedErrorsToSuppress, '/iframe'],
+  featureIndexPage: [],
 };
 
-Object.keys(pageTypes).forEach(pageType => {
```
Comment (simonsinclair) [inline a11y.js:82]: Please can you help me understand why this block has been refactored? Has the value returned by `getPageUrls` changed?

**Review by simonsinclair — COMMENTED** [review]
Excuse my ignorance! I have a few questions:

File: a11y.js (hunk 1 above)
Comment (karinathomasbbc) [reply a11y.js:77]: Makes the a11y tests (for STYs) pass locally.

File: a11y.js (hunk 2 above)
Comment (karinathomasbbc) [reply a11y.js:82]: I noticed that the tests weren't running properly if there were services / page types with multiple URLs - it was testing against a 404 page. This change was to ensure that each path returned by getPageUrls was tested individually.

Comment (simonsinclair) [reply a11y.js:82]: Ah, thanks for explaining, Karina.

**Review by simonsinclair — APPROVED** [review]

**Review by HarveyPeachey — APPROVED** [review]
LGTM. Good spot on the tests using multiple URL's using 404 pages 👍

**Pull request merged**